internal/resource/config/authenticationpolicies/settings/authentication_policies_settings_resource.go

```diff
@@ -0,0 +1,227 @@
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *authenticationPoliciesSettingsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan authenticationPoliciesSettingsModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createAuthenticationPoliciesSettings := client.NewAuthenticationPoliciesSettings()
+	err := addOptionalAuthenticationPoliciesSettingsFields(ctx, createAuthenticationPoliciesSettings, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for authentication policies settings", err.Error())
+		return
+	}
+	requestJson, err := createAuthenticationPoliciesSettings.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Add request: "+string(requestJson))
+	}
+
+	apiCreateAuthenticationPoliciesSettings := r.apiClient.AuthenticationPoliciesAPI.UpdateAuthenticationPolicySettings(config.ProviderBasicAuthContext(ctx, r.providerConfig))
+	apiCreateAuthenticationPoliciesSettings = apiCreateAuthenticationPoliciesSettings.Body(*createAuthenticationPoliciesSettings)
+	authenticationPoliciesSettingsResponse, httpResp, err := r.apiClient.AuthenticationPoliciesAPI.UpdateAuthenticationPolicySettingsExecute(apiCreateAuthenticationPoliciesSettings)
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while updating the authentication policies settings", err, httpResp)
+		return
+	}
+	responseJson, err := authenticationPoliciesSettingsResponse.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Add response: "+string(responseJson))
+	}
+
+	// Read the response into the state
+	var state authenticationPoliciesSettingsModel
+
+	readAuthenticationPoliciesSettingsResponse(ctx, authenticationPoliciesSettingsResponse, &state, nil)
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *authenticationPoliciesSettingsResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state authenticationPoliciesSettingsModel
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	apiReadAuthenticationPoliciesSettings, httpResp, err := r.apiClient.AuthenticationPoliciesAPI.GetAuthenticationPolicySettings(config.ProviderBasicAuthContext(ctx, r.providerConfig)).Execute()
+
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			config.ReportHttpErrorAsWarning(ctx, &resp.Diagnostics, "An error occurred while getting the authentication policies settings", err, httpResp)
+			resp.State.RemoveResource(ctx)
+		} else {
+			config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting the authentication policies settings", err, httpResp)
+		}
```

Comment:
Need a return after this line. Otherwise it will try to read into state from an empty response after a 404